src/auth/jwtclient.ts

```diff
@@ -107,46 +103,53 @@ export class JWT extends OAuth2Client {
   /**
    * Get the initial access token using gToken.
    * @param {function=} callback Optional callback.
+   * @returns Promise that resolves with credentials
    */
-  authorize(callback?: (err: Error|null, result: Credentials) => void) {
-    const done = callback || noop;
-    this.refreshToken(null, (err, result) => {
-      if (err) {
-        done(err);
-      } else if (!result) {
-        done(new Error('No result returned'));
-      } else {
-        this.credentials = result;
-        this.credentials.refresh_token = 'jwt-placeholder';
-        this.key = this.gtoken.key;
-        this.email = this.gtoken.iss;
-        done(null, result);
-      }
-    });
+  authorize(): Promise<Credentials>;
+  authorize(callback: (err: Error|null, result?: Credentials) => void): void;
+  authorize(callback?: (err: Error|null, result?: Credentials) => void):
+      Promise<Credentials>|void {
+    if (callback) {
+      this.authorizeAsync().then(r => callback(null, r)).catch(callback);
+    } else {
+      return this.authorizeAsync();
+    }
+  }
+
+  private async authorizeAsync(): Promise<Credentials> {
+    const result = await this.refreshToken();
+    if (!result) {
+      throw new Error('No result returned');
+    } else {
+      this.credentials = result.tokens;
+      this.credentials.refresh_token = 'jwt-placeholder';
+      this.key = this.gtoken.key;
+      this.email = this.gtoken.iss;
+      return result.tokens;
+    }
   }
 
   /**
    * Refreshes the access token.
    * @param {object=} ignored
-   * @param {function=} callback Optional callback.
    * @private
    */
-  refreshToken(
-      ignored: string|null,
-      callback?: (err: Error, credentials?: Credentials) => void) {
-    const done = callback || noop;
-    return this.createGToken((err, newGToken) => {
-      if (err) {
-        return done(err);
-      } else {
-        return newGToken.getToken((err2?: Error|null, token?: string|null) => {
-          return done(err2, {
+  async refreshToken(refreshToken?: string|null) {
```

Comment:
nit: drop the `async` unless you have `await`.

Reply:
Forgot to update this function after updating gtoken.  Now using await.